src/sensor/protocoldetector.cpp

```diff
@@ -190,18 +190,26 @@ bool ProtocolDetector::checkBuffer(const QByteArray& buffer)
     // since the buffer need to be clear for each try
     PingParserExt parser;
 
-    // Print information from detected devices
-    connect(&parser, &Parser::newMessage, this, [&](const ping_message& msg) {
-        common_device_information device_information(msg);
-        qCDebug(PING_PROTOCOL_PROTOCOLDETECTOR) << "Detect new device:"
-                                                << "\ndevice_revision:" << device_information.device_revision()
-                                                << "\nfirmware_version_major:" << device_information.firmware_version_major()
-                                                << "\nfirmware_version_minor:" << device_information.firmware_version_minor()
-                                                << "\nfirmware_version_patch:" << device_information.firmware_version_patch();
-    });
-
+    qCDebug(PING_PROTOCOL_PROTOCOLDETECTOR) << buffer;
     for (const auto& byte : buffer) {
         if(parser.parseByte(byte) == Parser::NEW_MESSAGE) {
+            // Print information from detected devices
+            common_device_information device_information(parser.rxMessage());
```

Comment:
To be safe, we should verify the message id before the copy. It might be a Nack or something unexpected.

Reply:
Ping1D will reply with a NACK message, that's why I'm not checking. Will see that I can do related to it, otherwise a TODO will be add;

Reply:
The suggestion is this: 
```
if (parser.rxMessage()->message_id() == CommonId::DEVICE_INFORMATION) {
    common_device_information device_information(parser.rxMessage());
    ...
}
```

You should not cast a pingmessage to a child type before checking the message id matches the child type.